src/snowflake/snowpark/catalog.py

```diff
@@ -0,0 +1,636 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import re
+from typing import List, NamedTuple, Optional, Union
+
+from snowflake.core import Root
+from snowflake.core.database import Database
+from snowflake.core.exceptions import NotFoundError
+from snowflake.core.function import Function
+from snowflake.core.procedure import Procedure
+from snowflake.core.schema import Schema
+from snowflake.core.table import Table
+from snowflake.core.user_defined_function import UserDefinedFunction
+from snowflake.core.view import View
+
+import snowflake.snowpark
+from snowflake.snowpark._internal.type_utils import convert_sp_to_sf_type
+from snowflake.snowpark.types import DataType
+
+
+class Column(NamedTuple):
```

Comment:
`snowflake.core` also has Column class. Should we reuse that one?

Reply:
let me check, maybe I can use that for my implementation.

Reply:
thanks. updated the PR